cssom-view-1/Overview.bs

```diff
@@ -420,22 +420,27 @@ an associated element or [=pseudo-element=] <var>element</var> and optionally a
 (which is "<code>auto</code>" if omitted),
 the following steps must be run:
 
-<ol>
- <li><a lt="smooth scroll aborted">Abort</a> any ongoing <a>smooth scroll</a> for <var>box</var>.
- <li>If the user agent honors the 'scroll-behavior' property and one of the following are true:
+1. <a lt="smooth scroll aborted">Abort</a> any ongoing <a>smooth scroll</a> for <var>box</var>.
+1. Resolve any pending promise for <var>box</var> with false.
+1. Let <var>scrollPromise</var> be a new promise.
+1. Return <var>scrollPromise</var>, and run the remaining steps <a>in parallel</a>.
+1. If the user agent honors the 'scroll-behavior' property and one of the following is true:
   <ul>
    <li><var>behavior</var> is "<code>auto</code>" and <var>element</var> is not null and its computed value of the
-   'scroll-behavior' property is ''scroll-behavior/smooth''
+   'scroll-behavior' property is ''scroll-behavior/smooth'', or
    <li><var>behavior</var> is <code>smooth</code>
   </ul>
-  ...then perform a <a>smooth scroll</a> of <var>box</var> to <var>position</var>. Once the position has finished updating, emit the <a event>scrollend</a> event.
-  Otherwise, perform an <a>instant scroll</a> of <var>box</var> to <var>position</var>. After an <a>instant scroll</a> emit the <a event>scrollend</a> event.
+  then perform a <a>smooth scroll</a> of <var>box</var> to <var>position</var>;
+  otherwise, perform an <a>instant scroll</a> of <var>box</var> to <var>position</var>.
+1. Wait until the position has finished updating.
+1. If the scroll position changed as a result of this call, emit the <a event>scrollend</a> event.
+1. Resolve <var>scrollPromise</var> with true if it is still in the pending state.
 
   Note: <code>behavior: "instant"</code> always performs an <a>instant scroll</a> by this algorithm.
 
   Note: If the scroll position did not change as a result of the user interaction or programmatic invocation, where no translations were applied as a result, then no <a event>scrollend</a> event fires because no scrolling occurred.
-</ol>
 
+Issue: What values to use when fulfilling programmatic scroll promises at scroll completion vs interruption? #12355
```

Comment:
It seems like it would make more sense to put this with the interface where we will currently say we return a boolean. Also can you make this a bit more descriptive, e.g.

Issue: Should we return a result object and if so what information should it provide? #12355